### PR TITLE
[9.0][MIG] medical prescription disease

### PR DESCRIPTION
**Pull request by lasley**
Odoo Medical Prescription Disease
=================================

Extension of medical that adds the disease concept into prescription order lines
in order to maintain a clean patient prescription history.

This is dependent on

- [X] #79

### CODE REVIEW

File: medical_prescription_disease/README.rst [hunk 1]
```diff
@@ -0,0 +1,59 @@
+.. image:: https://img.shields.io/badge/license-AGPL--3-blue.svg
+    :alt: License: AGPL-3
+
```
Comment (lasley) [inline medical_prescription_disease/README.rst:3]: Update readme template

File: medical_prescription_disease/__init__.py [hunk 2]
```diff
@@ -0,0 +1,6 @@
+# -*- coding: utf-8 -*-
+# Copyright 2016-2017 LasLabs Inc.
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from . import models
+from . import tests
```
Comment (lasley) [inline medical_prescription_disease/__init__.py:6]: Don't import tests

File: medical_prescription_disease/models/medical_prescription_order_line.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+class MedicalPrescriptionOrderLine(models.Model):
+    _inherit = 'medical.prescription.order.line'
+    disease_id = fields.Many2one(
+        string='Disease',
+        comodel_name='medical.patient.disease',
+        help='Disease diagnosis related to prescription.',
+    )
+
+    @api.multi
```
Comment (woodbrettm) [inline medical_prescription_disease/models/medical_prescription_order_line.py:16]: Should @api.onchange methods have an @api.multi? Confused as to why there's @api.multi, @api.onchange, and self.ensure_one() combined in this method.

File: medical_prescription_disease/views/medical_prescription_order_line_view.xml [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    Copyright 2016-2017 LasLabs Inc.
+    License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+-->
+
+<odoo>
+
+    <record id="medical_prescription_order_line_form_view" model="ir.ui.view">
+        <field name="name">Prescription Order Line</field>
+        <field name="model">medical.prescription.order.line</field>
+        <field name='inherit_id' ref='medical_prescription.medical_prescription_order_line_form_view'/>
```
Comment (woodbrettm) [inline medical_prescription_disease/views/medical_prescription_order_line_view.xml:13]: `"External ID not found in the system: medical_prescription.medical_prescription_order_line_form_view"`
Think that should be view_form

File: medical_prescription_disease/views/medical_prescription_order_line_view.xml [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+        <field name="arch" type="xml">
+            <xpath expr="//field[@name='patient_id']" position="after">
+                <field name="disease_id"
+                       context="{'default_patient_id': patient_id}"/>
+            </xpath>
+        </field>
+    </record>
+
+    <record id="medical_prescription_order_line_search_view" model="ir.ui.view">
```
Comment (woodbrettm) [inline medical_prescription_disease/views/medical_prescription_order_line_view.xml:22]: view_search

File: medical_prescription_disease/views/medical_prescription_order_line_view.xml [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+        <field name="name">Prescription Order Line Search</field>
+        <field name="model">medical.prescription.order.line</field>
+        <field name='inherit_id' ref='medical_prescription.medical_prescription_order_line_search_view'/>
```
Comment (woodbrettm) [inline medical_prescription_disease/views/medical_prescription_order_line_view.xml:25]: view_search

File: medical_prescription_disease/views/medical_prescription_order_line_view.xml [hunk 7]
```diff
@@ -0,0 +1,37 @@
+<?xml version="1.0" encoding="utf-8"?>
+
+<!--
+    Copyright 2016-2017 LasLabs Inc.
+    License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+-->
+
+<odoo>
+
+    <record id="medical_prescription_order_line_form_view" model="ir.ui.view">
```
Comment (woodbrettm) [inline medical_prescription_disease/views/medical_prescription_order_line_view.xml:10]: view_form

File: medical_prescription_disease/tests/test_medical_prescription_order_line.py [hunk 8]
```diff
@@ -0,0 +1,89 @@
+# -*- coding: utf-8 -*-
+# Copyright 2016-2017 LasLabs Inc.
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from openerp.tests.common import TransactionCase
+
+
+class TestMedicalPrescriptionOrderLine(TransactionCase):
```
Comment (woodbrettm) [inline medical_prescription_disease/tests/test_medical_prescription_order_line.py:8]: Do you think we should add demo data to this module? Seems like a lot of work in the test file for a single test, but I guess would also be a fair amount of work in demo as well, depending on the required fields.

File: medical_prescription_disease/models/medical_prescription_order_line.py [hunk 9]
```diff
@@ -0,0 +1,33 @@
+# -*- coding: utf-8 -*-
+# Copyright 2016-2017 LasLabs Inc.
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
```
Comment (woodbrettm) [inline medical_prescription_disease/models/medical_prescription_order_line.py:3]: Should the licenses be LGPL-3 now, or is AGPL-3 still correct?

File: medical_prescription_disease/views/medical_patient_disease_view.xml [hunk 10]
```diff
@@ -0,0 +1,29 @@
+<?xml version="1.0" encoding="utf-8"?>
+
+<!--
+    Copyright 2016-2017 LasLabs Inc.
+    License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
```
Comment (woodbrettm) [inline medical_prescription_disease/views/medical_patient_disease_view.xml:5]:
````xml
<?xml version="1.0" encoding="utf-8"?>
<!-- Copyright 2016-2017 LasLabs Inc.
     License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl). -->
````
or lgpl not sure :D

File: medical_prescription_disease/views/medical_prescription_order_line_view.xml [hunk 11]
```diff
@@ -0,0 +1,37 @@
+<?xml version="1.0" encoding="utf-8"?>
+
+<!--
+    Copyright 2016-2017 LasLabs Inc.
+    License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+-->
```
Comment (woodbrettm) [inline medical_prescription_disease/views/medical_prescription_order_line_view.xml:6]:
````xml
<?xml version="1.0" encoding="utf-8"?>
<!-- Copyright 2016-2017 LasLabs Inc.
     License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl). -->
````
or lgpl not sure :D

File: medical_prescription_disease/models/medical_patient_disease.py [hunk 12]
```diff
@@ -0,0 +1,15 @@
+# -*- coding: utf-8 -*-
+# Copyright 2016-2017 LasLabs Inc.
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from openerp import models, fields
+
+
+class MedicalPatientDisease(models.Model):
+    _inherit = 'medical.patient.disease'
```
Comment (woodbrettm) [inline medical_prescription_disease/models/medical_patient_disease.py:9]: should be empty line between this, fields, and class declaration.

File: medical_prescription_disease/__openerp__.py [hunk 13]
```diff
@@ -0,0 +1,23 @@
+# -*- coding: utf-8 -*-
+# Copyright 2016-2017 LasLabs Inc.
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+{
```
Comment (woodbrettm) [inline medical_prescription_disease/__openerp__.py:5]: Some keys are ordered incorrectly as per [OCA](https://github.com/OCA/maintainer-tools/blob/master/template/module/__openerp__.py)

**Review by woodbrettm — CHANGES_REQUESTED** [review]
Couple things but other than that looks pretty good!

**Comment by lasley** [comment]
Closing - we no longer have interest in v9. We will continue these forward in v10.

Anyone is welcome to pick the code up and reintroduce a PR, however.

**Pull request closed without merging**